File: src/main/java/com/kmatrokhin/uvbot/chatgpt/ChatResponse.java
package com.kmatrokhin.uvbot.chatgpt;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
@ToString
public class ChatResponse {
    private List<Choice> choices;

    @Getter
    @Setter
    @JsonIgnoreProperties(ignoreUnknown = true)
    @ToString
    public static class Choice {
        private int index;
        private Message message;
    }
}

File: src/main/java/com/kmatrokhin/uvbot/chatgpt/ChatResponse.kt
package com.kmatrokhin.uvbot.chatgpt

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class ChatResponse(
    val choices: List<Choice>
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Choice(
        val index: Int = 0,
        val message: Message
    )
}
